src/gui/qgsqueryresultwidget.cpp

```diff
@@ -111,6 +106,38 @@ QgsQueryResultWidget::~QgsQueryResultWidget()
   cancelRunningQuery();
 }
 
+void QgsQueryResultWidget::setSqlVectorLayerOptions( const QgsAbstractDatabaseProviderConnection::SqlVectorLayerOptions &options )
+{
+  mSqlVectorLayerOptions = options;
+  if ( ! options.sql.isEmpty() )
+  {
+    setQuery( options.sql );
+  }
+  mAvoidSelectingAsFeatureIdCheckBox->setChecked( options.disableSelectAtId );
+  mPkColumnsCheckBox->setChecked( ! options.primaryKeyColumns.isEmpty() );
+  mPkColumnsComboBox->setCheckedItems( {} );
+  if ( ! options.primaryKeyColumns.isEmpty() )
+  {
+    mPkColumnsComboBox->setCheckedItems( options.primaryKeyColumns );
+  }
+  mGeometryColumnCheckBox->setChecked( ! options.geometryColumn.isEmpty() );
+  mGeometryColumnComboBox->clear();
+  if ( ! options.geometryColumn.isEmpty() )
+  {
+    mGeometryColumnComboBox->setCurrentText( options.geometryColumn );
+  }
+  mFilterLineEdit->setText( options.filter );
+  mLayerNameLineEdit->setText( options.layerName );
+}
+
+void QgsQueryResultWidget::setUpdateSqlLayerMode( bool updateMode )
+{
+  mUpdateSqlLayerMode = updateMode;
+  mLoadAsNewLayerGroupBox->setTitle( updateMode ? tr( "Update query layer" ) : tr( "Load as new layer" ) );
+  mLoadLayerPushButton->setText( updateMode ? tr( "Update layer" ) : tr( "Load layer" ) );
```

Comment:
```suggestion
  mLoadAsNewLayerGroupBox->setTitle( updateMode ? tr( "Update Query Layer" ) : tr( "Load as New Layer" ) );
  mLoadLayerPushButton->setText( updateMode ? tr( "Update Layer" ) : tr( "Load Layer" ) );
```

Reply:
I've added an ENUM for the widget mode, I can foresee there might be more modes in the future.